gossip.asciidoc

```diff
@@ -644,68 +644,388 @@ channel graph) has the following attributes:
+importantly, the `signature` MUST be valid, and the `timestamp` MUST always be
+increasing in order to invalidate prior updates.
+
+##### Channel & Message Flags
+
+As mentioned earlier, there're two `channel_update` messages for each direction
+of the channel. Therefore, we need to be able to identify which direction of
+the channel is being updated. If we have two nodes Alice, and Bob, then there
+may exist a channel update in the direction of Alice -> Bob, as well as for Bob
+-> Alice. In the first case, we require Alice to be the one that signs the
+update (as she will be carrying the HTLC), and in the second case, we require
+that Bob is the one that signs the update.
+
+Thankfully, we can use a single bit to denote which direction is sending out
+the update. Recall that in the `channel_announcement` message, we sort the node
+public keys of both nodes lexicographically and refer to the "first" node as
+`node_1`, and the second as `node_2`. We then carry this over to the
+`channel_update` message within the `channel_flags` field. The field itself is
+a bit field, with the LSB denoting the "direction" of the channel update. If
+the bit is set, then this update belongs to `node_1`, otherwise to `node_2`.
+
+##### Routing Policy Control in Channel Updates
+
+The main purpose of the `channel_update` message is to allow nodes to be able
+to update their routing policy on the network. As conditions change, nodes may
+want to update aspects of their routing policy that dictate what can route
+through their channels, and how the HTLC that arrive at their channels should
+look like in order to be forwarded along.
+
+First, routing fees. In the LN, routing fees are the difference between the
+outgoing and incoming HTLCs of a forwarded payment. By sending out less than
+they receive, routing nodes are able to profit by earning a fees for their help
+to route the HTLC through the network. All HTLCs in the LN are expressed in
+`msat` or milli-satoshis instead of satoshis or Bitcoin. We use `msat`, as it
+allows us to charge fees for 1 sat payments. The routing fees has two
+component, `fee_base_msat` and `fee_proportional_millionths`. In order to
+calculate how much a payment of `N msat` should pay to route through a channel,
+we use the following equation:
+```
+fee = fee_base_msat + (N * fee_proportional_millionths) / 1,000,000
+```
 
-// TODO(roasbeef): rename to "the structure of the channel graph"?
+This may look a bit strange at first, but all we're doing in the latter half of
+the equation is computing the rate to be paid. Rather than floating point
+arithmetic, we use fixed point arithmetic with 1 million as our base. This lets
+us express very small amounts of fees to be paid in the `mast` unit. Let's say
+we use a value of `100,000` for `fee_proportional_millionths`, this is
+equivalent to a `10%` fee as `100,000/1,000,000 = 0.10` or `10%.
+
+Another important security related parameter for a channel is the
```

Comment:
I would move this somewhere. Either we move it to the security chapter or routing or make it an info box. 

In any case I think we should also create an example where we show with specific timelocks and blockheights / mempool what could happen. 

I think something like this is one of the more crucial things for a student to understand thus my suggestion to heavily elaborate on this